### PR TITLE
issue: HPCINFRA-1798 [CI] clang 15 support

### PR DESCRIPTION
**Pull request by vialogi**
## Description
We're requested to provide clang-15 support for Style step in CI

##### What
Switch to ubuntu22.04-based container for Style step and provide clang-15 support

##### Why ?
As per the dev request

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

**Comment by vialogi** [comment]
bot:retest

**Review by dpressle — CHANGES_REQUESTED** [review]
Why i see changes related to Doca? it has nothing to do with this task

**Comment by vialogi** [comment]
> Why i see changes related to Doca? it has nothing to do with this task

Removed and made the changes we discussed

File: .ci/dockerfiles/Dockerfile.ubuntu22.04 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM harbor.mellanox.com/swx-infra/media/ubuntu_22.04:mofed-5.7-0.1.9.0 AS style
+RUN apt-get update \
+ && apt-get install -y clang-15 clang-format-15 \
+ && update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-15 100 \
+ && update-alternatives --install /usr/bin/clang clang /usr/bin/clang-15 100 \
+                    --slave /usr/bin/clang++ clang++ /usr/bin/clang++-15 \
+ && apt-get clean && rm -rf /var/lib/apt/lists/*
+
```
Comment (dpressle) [inline .ci/dockerfiles/Dockerfile.ubuntu22.04:8]: missing newline

File: .ci/dockerfiles/Dockerfile.ubuntu22.04 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM harbor.mellanox.com/swx-infra/media/ubuntu_22.04:mofed-5.7-0.1.9.0 AS style
```
Comment (dpressle) [inline .ci/dockerfiles/Dockerfile.ubuntu22.04:1]: why we need ofed image?

**Review by dpressle — CHANGES_REQUESTED** [review]

File: .ci/dockerfiles/Dockerfile.ubuntu22.04 (hunk 2 above)
Comment (vialogi) [reply .ci/dockerfiles/Dockerfile.ubuntu22.04:1]: for DPCP, which compilation is triggered in globals.sh
maybe for static tests, we need to change the approach and not compile anything

File: .ci/dockerfiles/Dockerfile.ubuntu22.04 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM harbor.mellanox.com/swx-infra/media/base_ubuntu_22.04 AS style
+RUN apt-get update \
+ && apt-get install -y clang-15 clang-format-15 \
+ && update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-15 100 \
+ && update-alternatives --install /usr/bin/clang clang /usr/bin/clang-15 100 \
+                    --slave /usr/bin/clang++ clang++ /usr/bin/clang++-15 \
+ && apt-get clean && rm -rf /var/lib/apt/lists/*
+
```
Comment (dpressle) [inline .ci/dockerfiles/Dockerfile.ubuntu22.04:8]: missing newline

File: .ci/matrix_job.yaml [hunk 4]
```diff
@@ -184,10 +186,11 @@ steps:
   - name: Style
     enable: ${do_style}
     containerSelector:
-      - "{name: 'toolbox', category: 'tool'}"
+      - "{name: 'style', category: 'tool'}"
     agentSelector:
       - "{nodeLabel: 'skip-agent'}"
     run: |
+      export COMPILE_DPCP=false
```
Comment (dpressle) [inline .ci/matrix_job.yaml:193]: i prefer not to do such change now if its not needed (even if current behavior is wrong) can we remove it and keep build dpcp ?

File: contrib/jenkins_tests/style.sh [hunk 5]
```diff
@@ -4,14 +4,16 @@ source $(dirname $0)/globals.sh
...
-
 cd $WORKSPACE
 
 rm -rf $style_dir
 mkdir -p $style_dir
 cd $style_dir
 
+echo "Check the clang/clang-format version:"
+echo "clang: $(clang --version |grep -i version)"
+echo "clang-format: $(clang-format --version |grep -i version)"
```
Comment (dpressle) [inline contrib/jenkins_tests/style.sh:15]: What about verifying clang version and failing if we have wrong version ?

**Review by dpressle — CHANGES_REQUESTED** [review]

**Comment by vialogi** [comment]
bot:retest

**Comment by vialogi** [comment]
bot:retest

**Comment by pasis** [comment]
@vialogi, there is ticket number mismatch between PR name and commit messages

**Comment by pasis** [comment]
Merged as part of #200, thanks!

**Pull request closed without merging**